### PR TITLE
ref(craft): Modernize Craft config

### PR DESCRIPTION
**Pull request by BYK**

### CODE REVIEW

**Review by rhcarvalho — COMMENTED** [review]
I haven't followed all changes to craft, @BYK would you mind explaining briefly what's going on here for context?

File: .craft.yml [hunk 1]
```diff
@@ -1,18 +1,12 @@
-minVersion: '0.9.2'
-github:
-  owner: getsentry
-  repo: sentry-go
+minVersion: 0.23.1
 preReleaseCommand: bash scripts/craft-pre-release.sh
 changelogPolicy: simple
-statusProvider:
-  name: github
```
[the remark this answers is not in the document]
Comment (BYK) [reply .craft.yml:8]: Zeus is now gone and the default is already `github`.

File: .craft.yml [hunk 2]
```diff
@@ -1,18 +1,12 @@
...
 targets:
   - name: github
     includeNames: /none/
     tagPrefix: v
   - name: registry
-    type: sdk
-    config:
-      canonical: "github:getsentry/sentry-go"
+    sdks:
+      github:getsentry/sentry-go:
```
[the remark this answers is not in the document]
Comment (BYK) [reply .craft.yml:12]: Yup.

[the remark this answers is not in the document]
Comment (rhcarvalho) [reply .craft.yml:12]: 👍 

Out of curiosity, what kind non-null value would this take?

**Review by rhcarvalho — APPROVED** [review]

File: .craft.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (BYK) [reply .craft.yml:12]: You can have a `checksums` field or a `urlTemplate` for apps for example. Here's an example from Craft itself: https://github.com/getsentry/craft/blob/e5fad6162b57848f7ce8cd4228adbbab9b185052/.craft.yml#L18-L24

**Pull request merged**